### PR TITLE
feat(packages/tuono): config – add `optimizeDeps` and `plugins` to `vite` options

### LINKED ISSUE #183: [Refactor]: Remove `mdx` references from `tuono` npm package
There are some `vite` related config added in the `tuono` npm package referring to `@mdx-js/react`.
With the addition of the new `tuono.config.ts` they should be moved there rather than being in the source code of the package

### PR DESCRIPTION
**Pull request by marcalexiei**
## Context & Description

Close #183.

Remove `mdx` related config from `tuono` npm package

---

- [x] Remove `console.log`
- [x] Remove test text in mdx example
- [x] Refine config unit tests
- [x] Remove `include` option from `vite` `react` plugin 
- [x] Create followup task for documentation config update 
        (#212)

### CODE REVIEW

File: packages/tuono/src/build/index.ts [hunk 1]
```diff
@@ -3,86 +3,102 @@ import { build, createServer, mergeConfig } from 'vite'
...
-    // @ts-expect-error: TS configuration issue.
-    react({ include: /\.(jsx|js|mdx|md|tsx|ts)$/ }),
-    ViteFsRouter(),
-    LazyLoadingPlugin(),
-  ],
+/**
+ * From a given {@link TuonoConfig} return a `vite` "mergeable" {@link InlineConfig}
+ * including all default tuono related options
+ */
+function createBaseViteConfigFromTuonoConfig(
```
Comment (Valerioageno) [inline packages/tuono/src/build/index.ts:17]: Super cool!

File: packages/tuono/src/build/index.ts [hunk 2]
```diff
@@ -3,86 +3,102 @@ import { build, createServer, mergeConfig } from 'vite'
...
-          noExternal: true,
-        },
-      }),
+      ),
     )
   })
 }
 
 const developmentCSRWatch = (): void => {
+  console.info('developmentCSRWatch')
```
Comment (Valerioageno) [inline packages/tuono/src/build/index.ts:79]: I'd remove this to leave all the logs on rust's side.

**Review by Valerioageno — CHANGES_REQUESTED** [review]
The PR looks amazing. I left a very little comment about a `console.log`.

File: packages/tuono/src/build/index.ts (hunk 2 above)
Comment (marcalexiei) [reply packages/tuono/src/build/index.ts]: An oversight when I removed the debug logs 😩

**Review by Valerioageno — APPROVED** [review]
LGTM. Thanks

**Pull request merged**